examples/C++/gstreamer/gst-snowboy/gst-plugin/src/gstsnowboy.cc

```diff
@@ -0,0 +1,415 @@
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Alternatively, the contents of this file may be used under the
+ * GNU Lesser General Public License Version 2.1 (the "LGPL"), in
+ * which case the following provisions apply instead of the ones
+ * mentioned above:
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+/**
+ * SECTION:element-plugin
+ *
+ * Snowboy hotword detector gst plugin.  Uses the snowboy detector library
+ * and links results to the `hotword-detect` signal which should be monitored
+ * from your application to determine when a hotword has been found.
+ *
+ * The plugin allows multiple models to be used at the same time using the
+ * `models` property.  It should be a comma-separated list of model files e.g.,
+ * models=file1.umdl,file2.umdl.
+ *
+ * The hotword-detect signal will provide an integer which is the model index e.g.,
+ * 0=>file1, 1=>file2 etc.
+ *
+ * The plugin is designed to work in a larger pipeline structure that could be
+ * used to perform additional audio processing.  For this reason the input data
+ * on the sinkpad is also passed through to the srcpad.  If you don't need this
+ * then just terminate with a `fakesink` element.
+ *
+ * The plugin only works with single channel audio S16 @ 16KHz as reflected by
+ * the fixed caps.
+ *
+ * Note that the snowboy detector is a C++ library which means we have to compile
+ * and link this plugin using a C++ compiler.  The gstreamer deps are wrapped in
+ * an extern "C" declaration so that the boilerplate code compiles clean.
+ *
+ * <refsect2>
+ * <title>Example launch line</title>
+ * |[
+ * gst-launch -v -m pulsesrc ! snowboy ! fakesink
+ * ]|
+ * </refsect2>
+ */
+ 
+extern "C" {
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <gst/gst.h>
+#include <gst/audio/audio.h>
+#include <gst/audio/gstaudiofilter.h>
+#include <string.h>
+
+}
+
+#include "snowboy-detect.h"
+
+GST_DEBUG_CATEGORY_STATIC (snowboy_debug);
+#define GST_CAT_DEFAULT snowboy_debug
+
+#define GST_TYPE_SNOWBOY (gst_snowboy_get_type())
+G_DECLARE_FINAL_TYPE (GstSnowboy, gst_snowboy,
+    GST, SNOWBOY, GstAudioFilter)
+
+struct _GstSnowboy
+{
+  GstAudioFilter audiofilter;
```

Comment:
Can we move the "{" right after the previous line? For example:

`struct _GstSnowboy {`

so that it's consistent with the rest of the code?

Also, let's try to keep the line less than 80 characters.